src/test/java/com/xceptance/neodymium/tests/NeodymiumTest.java

```diff
@@ -52,29 +54,106 @@ public static void deleteTempFile(File tempFile)
     }
 
     public void check(Result result, boolean expectedSuccessful, int expectedRunCount, int expectedIgnoreCount, int expectedFailCount,
-                      String expectedFailureMessage)
+                      Map<String, String> expectedFailureMessages)
     {
-        Assert.assertEquals("Test successful", expectedSuccessful, result.wasSuccessful());
-        Assert.assertEquals("Method run count", expectedRunCount, result.getRunCount());
-        Assert.assertEquals("Method ignore count", expectedIgnoreCount, result.getIgnoreCount());
-        Assert.assertEquals("Method fail count", expectedFailCount, result.getFailureCount());
-
-        if (expectedFailureMessage != null)
+        String stackTrace = "";
+        for (Failure failue : result.getFailures())
+        {
+            stackTrace += failue.getTrace();
```

Comment:
Please either use class StringBuilder for assembling strings
```java
final String stackTrace;
if(result.getFailureCount() > 0)
{
  final StringBuilder sb = new StringBuilder(1024);
  for(final Failure failure : result.getFailures())
  {
    sb.append(failure.getTrace());
  }
  stackTrace = sb.toString();
}
else
{
  stackTrace = "n/a";
}
```
or use streams to compute the final stack trace
```java
final Optional<String> accumulatedTrace = result.getFailures().stream().map(Failure::getTrace).reduce(String::concat);
final String stackTrace = accumulatedTrace.orElseGet("n/a");
```

Reply:
Done